app/src/main/java/com/example/androidapp/ui/home/HomeFragment.java

```diff
@@ -96,37 +123,144 @@ public void startAnimation(RelativeLayout area, GifImageView chic) {
             public void run() {
                 if (activity != null) {
                     activity.runOnUiThread( ()-> {
-                        RelativeLayout.LayoutParams layoutParams = new RelativeLayout.LayoutParams(120, 120);
-
-                        if (animation_frame == 0) {
-                            // random the direction, -1 -> left, 0 -> remain, 1 -> right
-                            next_x_direction = new Random().nextInt(3) + -1;
-                            next_y_direction = new Random().nextInt(3) + -1;
-                            // constrain chic walking area, if chic is out of range, then change direction
-                            if (x_chic > 190) next_x_direction = -1;
-                            if (x_chic < 50) next_x_direction = 1;
-                            if (y_chic > 110) next_y_direction = -1;
-                            if (y_chic < 50) next_y_direction = 1;
-                            // horizontally flip the img based on direction
-                            if (next_x_direction != 0) chic.setScaleX(-next_x_direction);
-                        }
+                        if (chic_state == "idle") {
```

Comment:
You will need to comment and document better this part of code.

e.g. Describe why you need to check the chick state.

However it is enough to do it after the 20th and before the 23rd.